### PR TITLE
Userhistory data

### PR DESCRIPTION
**Pull request by davidfurey**
Get users recent activity from the userhistory api instead of relying on the client posting this data

### CODE REVIEW

File: recommender/app/controllers/Application.scala [hunk 1]
```diff
@@ -51,6 +57,29 @@ object Application extends Controller {
...
+    browserId: String,
+    webPublicationDate: Option[DateRangeFilter],
+    tags: Option[QueryBoost],
+    disableDateFilter: Option[Boolean],
+    pageSize: Option[Int]
+    ) = Action.async {
+    val dateFilter =
+      webPublicationDate
+        .orElse(Some(defaultDateRangeFilter))
+        .filterNot(_ => disableDateFilter.contains(true))
```
Comment (alexduf) [inline recommender/app/controllers/Application.scala]: It took me a while to read this one. Would you mind making it a bit clearer? I suggest something like that:

``` scala
disableDateFilter match {
  case Some(true) => None
  case _ => webPublicationDate.getOrElse(defaultDateRangeFilter)
}
```

File: recommender/conf/routes [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # This file defines all application routes (Higher priority routes first)
 # ~~~~
 
-GET        /healthcheck                     controllers.Application.healthCheck
+GET        /healthcheck                             controllers.Application.healthCheck
 
-# GET         /recommendations/:userId        controllers.Application.recommendationsFromUserId(userId, webPublicationDate: Option[DateRangeFilter], tags: Option[QueryBoost], disableDateFilter: Option[Boolean], pageSize: Option[Int])
```
Comment (alexduf) [inline recommender/conf/routes:7]: is it expected that recommendationsFromUserId disapears? I guess it's not used ?

Comment (alexduf) [inline recommender/conf/routes:7]: Nevermind, I missed the #

**Comment by alexduf** [comment]
aside from readability comment, it looks good to me :+1:

**Comment by agulowaty** [comment]
:+1:

**Pull request merged**